components/group/groupSlice.ts

```diff
@@ -108,9 +147,68 @@ export const groupSlice = createSlice({
             console.log(`Create user rejected: ${JSON.stringify(action.error)}`)
             state.loading = false
         })
+        // createNewGroup
+        builder.addCase(fetchRegisteredGroups.pending, (state) => {
+            state.loading = true
+        })
+        builder.addCase(fetchRegisteredGroups.fulfilled, (state, action) => {
+            state.loading = false
+            state.groups = action.payload
+        })
+        builder.addCase(fetchRegisteredGroups.rejected, (state, action) => {
+            console.log(`Create user rejected: ${JSON.stringify(action.error)}`)
+            state.loading = false
+        })
     },
 })
 
 export const groupListSelector = (state: RootState) => state.group.groups
 
 export default groupSlice.reducer
+
+const registeredGroupsSelector = (state: RootState) => state.group.registeredGroups
+
+export const useGroupStorage: () => [RegisteredGroup[], (groups: RegisteredGroup[]) => void] = () => {
+    const registeredGroups = useAppSelector(registeredGroupsSelector)
+    const dispatch = useAppDispatch()
+    const setRegisteredGroups = (registeredGroups: RegisteredGroup[]) => {
+        dispatch(groupSlice.actions.registerNewGroups(registeredGroups))
+    }
```

Comment:
feels like this should have a stable reference created with a `useCallback` hook.